system-tests/test/deposit-redemption.test.ts

```diff
@@ -372,4 +374,276 @@ describe("System Test - Deposit and redemption", () => {
+        Deposit swept on Bitcoin chain:
+        - Transaction hash: ${sweepUtxo.transactionHash}
+      `)
+
+        // Unlike in the deposit transaction case, we must wait for the sweep
+        // transaction to have an enough number of confirmations. This is
+        // because the bridge performs the SPV proof of that transaction.
+        await waitTransactionConfirmed(
+          electrumClient,
+          sweepUtxo.transactionHash
+        )
+
+        await fakeRelayDifficulty(
+          relay,
+          electrumClient,
+          sweepUtxo.transactionHash
+        )
+
+        // TODO: Consider fetching the current wallet main UTXO and passing it
+        //       here. This will allow running this test scenario multiple
+        //       times for the same wallet.
+        await SpvMaintainer.submitDepositSweepProof(
+          sweepUtxo.transactionHash,
+          // This is the first sweep of the given wallet so there is no main UTXO.
+          {
+            // The function expects an unprefixed hash.
+            transactionHash: BitcoinTransactionHash.from(constants.HashZero),
+            outputIndex: 0,
+            value: BigNumber.from(0),
+          },
+          maintainerBridgeHandle,
+          electrumClient
+        )
+
+        console.log(`
+        Deposit sweep proved on the bridge
+      `)
+      })
+
+      it("should broadcast the sweep transaction on the Bitcoin network", async () => {
+        expect(
+          (await electrumClient.getRawTransaction(sweepUtxo.transactionHash))
+            .transactionHex.length
+        ).to.be.greaterThan(0)
+      })
+
+      it("should sweep the deposit on the bridge", async () => {
+        const { sweptAt } = await TBTC.getRevealedDeposit(
+          depositUtxo,
+          maintainerBridgeHandle
+        )
+        expect(sweptAt).to.be.greaterThan(0)
+      })
+
+      it("should increase depositor's balance in the bank", async () => {
+        const { treasuryFee } = await TBTC.getRevealedDeposit(
+          depositUtxo,
+          maintainerBridgeHandle
+        )
+
+        const expectedBalance = depositAmount
+          .sub(treasuryFee)
+          .sub(depositSweepTxFee)
+
+        const actualBalance = await bank.balanceOf(
+          systemTestsContext.depositor.address
+        )
+
+        expect(actualBalance).to.be.equal(expectedBalance)
+      })
+
+      context("when redemption is requested", () => {
+        let requestedAmount: BigNumber
+        let redeemerOutputScript: string
+        let redemptionRequest: RedemptionRequest
+
+        before("request the redemption", async () => {
+          // Redeem the full depositor's balance.
+          requestedAmount = await bank.balanceOf(
+            systemTestsContext.depositor.address
+          )
+
+          // Allow the bridge to take the redeemed bank balance.
+          await bank
+            .connect(systemTestsContext.depositor)
+            .approveBalance(bridgeAddress, requestedAmount)
```

Comment:
Shouldn't we get the balance from tBTC token contract if we used the TBTCVault contract address during the reveal?

Reply:
Yes. I removed:
```
          await bank
            .connect(systemTestsContext.depositor)
            .approveBalance(bridgeAddress, requestedAmount)
```
Since the depositor request redemption via `TBTC.requestRedmption(...)` we don't need the above code.